test/e2e/node_reuse_test.go

```diff
@@ -137,93 +139,79 @@ func nodeReuse(clusterClient client.Client) {
 	Expect(err).To(BeNil(), "Failed to patch KubeadmControlPlane")
 
 	By("Check if only a single machine is in Deleting state and no other new machines are in Provisioning state")
-	Eventually(
-		func(g Gomega) {
-			machines := &clusterv1.MachineList{}
-			g.Expect(clusterClient.List(ctx, machines, client.InNamespace(namespace))).To(Succeed())
-			deletingCount := 0
-			for _, machine := range machines.Items {
-				Expect(machine.Status.GetTypedPhase() == clusterv1.MachinePhaseProvisioning).To(BeFalse()) // Ensure no machine is provisioning
-				if machine.Status.GetTypedPhase() == clusterv1.MachinePhaseDeleting {
-					deletingCount++
-				}
-			}
-			g.Expect(deletingCount).To(Equal(1))
-		}, e2eConfig.GetIntervals(specName, "wait-machine-deleting")...,
-	).Should(Succeed())
+	waitForNumMachinesInState(ctx, clusterv1.MachinePhaseDeleting, waitForNumInput{
+		Client:    clusterClient,
+		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Replicas:  1,
+		Intervals: e2eConfig.GetIntervals(specName, "wait-machine-deleting"),
+	})
+	// Since we do scale in, no Machine should start provisioning yet (the old must be deleted first)
+	machineList := &clusterv1.MachineList{}
+	Expect(clusterClient.List(ctx, machineList, client.InNamespace(namespace))).To(Succeed())
+	Expect(filterMachinesByPhase(machineList.Items, clusterv1.MachinePhaseProvisioning)).To(HaveLen(0))
 
```

Comment:
I think it is quite safe like this. If it would start provisioning a machine that should happen immediately. But we can add a Consistently call also, the only question is for how long? If we wait for too long the deleting machine will start provisioning again!
Do you have a suggestion for how long would be good to wait?

Reply:
agree I guess the time needed for a machine to become deleting is almost the same needed to become provisioning so the the extra delay should relatively small than the average need to machine to change state we can measure that (expect to be less than 30s) in that case I would suggest 10s?